contrib/google.golang.org/grpc/grpc_test.go

```diff
@@ -663,3 +664,111 @@ func TestIgnoredMethods(t *testing.T) {
 		}
 	})
 }
+
+func TestWithMetadataTags(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	rig, err := newRig(true, WithMetadataTags())
+	if err != nil {
+		t.Fatalf("error setting up rig: %s", err)
+	}
+	defer rig.Close()
+
+	ctx := context.Background()
+	ctx = metadata.AppendToOutgoingContext(ctx, "test-key", "test-value")
+	span, ctx := tracer.StartSpanFromContext(ctx, "x", tracer.ServiceName("y"), tracer.ResourceName("z"))
+	rig.client.Ping(ctx, &FixtureRequest{Name: "pass"})
+	span.Finish()
+
+	spans := mt.FinishedSpans()
+
+	var serverSpan mocktracer.Span
+
+	for _, s := range spans {
+		switch s.OperationName() {
+		case "grpc.server":
+			serverSpan = s
+		}
+	}
+
+	span, _ = tracer.SpanFromContext(ctx)
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-trace-id")
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-parent-id")
+	assert.NotContains(t, serverSpan.Tags(), tagMetadataPrefix+"x-datadog-sampling-priority")
+	assert.Equal(t, serverSpan.Tag(tagMetadataPrefix+"test-key"), []string{"test-value"})
+}
+
+func TestIgnoredMetadata(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+	for _, c := range []struct {
+		ignore []string
+		exp    int
+	}{
+		{ignore: []string{}, exp: 1},
+		{ignore: []string{tagMetadataPrefix + "unknown"}, exp: 1},
+		{ignore: []string{tagMetadataPrefix + "test-key"}, exp: 0},
+	} {
+		rig, err := newRig(true, WithMetadataTags(), WithIgnoredMetadata(c.ignore...))
+		if err != nil {
+			t.Fatalf("error setting up rig: %s", err)
+		}
+		ctx := context.Background()
+		ctx = metadata.AppendToOutgoingContext(ctx, "test-key", "test-value")
+		span, ctx := tracer.StartSpanFromContext(ctx, "x", tracer.ServiceName("y"), tracer.ResourceName("z"))
+		rig.client.Ping(ctx, &FixtureRequest{Name: "pass"})
+		span.Finish()
+
+		spans := mt.FinishedSpans()
+
+		var serverSpan mocktracer.Span
+
```

Comment:
Can we remove this empty line? We generally try not to have too many in Go. I noticed this is the same in the standard library and other projects.